packages/contracts-ts/scripts/generate-addresses.ts

```diff
@@ -0,0 +1,254 @@
+    introduced: 'Legacy',
+    deprecated: true,
+    proxied: false,
+  },
+  WETH9: {
+    address: '0x4200000000000000000000000000000000000006',
+    introduced: 'Legacy',
+    deprecated: false,
+    proxied: false,
+  },
+  L2CrossDomainMessenger: {
+    address: '0x4200000000000000000000000000000000000007',
+    introduced: 'Legacy',
+    deprecated: false,
+    proxied: true,
+  },
+  L2StandardBridge: {
+    address: '0x4200000000000000000000000000000000000010',
+    introduced: 'Legacy',
+    deprecated: false,
+    proxied: true,
+  },
+  SequencerFeeVault: {
+    address: '0x4200000000000000000000000000000000000011',
+    introduced: 'Legacy',
+    deprecated: false,
+    proxied: true,
+  },
+  OptimismMintableERC20Factory: {
+    address: '0x4200000000000000000000000000000000000012',
+    introduced: 'Legacy',
+    deprecated: false,
+    proxied: true,
+  },
+  L1BlockNumber: {
+    address: '0x4200000000000000000000000000000000000013',
+    introduced: 'Legacy',
+    deprecated: true,
+    proxied: true,
+  },
+  GasPriceOracle: {
+    address: '0x420000000000000000000000000000000000000F',
+    introduced: 'Legacy',
+    deprecated: false,
+    proxied: true,
+  },
+  GovernanceToken: {
+    address: '0x4200000000000000000000000000000000000042',
+    introduced: 'Legacy',
+    deprecated: false,
+    proxied: false,
+  },
+  L1Block: {
+    address: '0x4200000000000000000000000000000000000015',
+    introduced: 'Bedrock',
+    deprecated: false,
+    proxied: true,
+  },
+  L2ToL1MessagePasser: {
+    address: '0x4200000000000000000000000000000000000016',
+    introduced: 'Bedrock',
+    deprecated: false,
+    proxied: true,
+  },
+  L2ERC721Bridge: {
+    address: '0x4200000000000000000000000000000000000014',
+    introduced: 'Legacy',
+    deprecated: false,
+    proxied: true,
+  },
+  OptimismMintableERC721Factory: {
+    address: '0x4200000000000000000000000000000000000017',
+    introduced: 'Bedrock',
+    deprecated: false,
+    proxied: true,
+  },
+  ProxyAdmin: {
+    address: '0x4200000000000000000000000000000000000018',
+    introduced: 'Bedrock',
+    deprecated: false,
+    proxied: true,
+  },
+  BaseFeeVault: {
+    address: '0x4200000000000000000000000000000000000019',
+    introduced: 'Bedrock',
+    deprecated: false,
+    proxied: true,
+  },
+  L1FeeVault: {
+    address: '0x420000000000000000000000000000000000001a',
+    introduced: 'Bedrock',
+    deprecated: false,
+    proxied: true,
+  },
+} as const
+
+const chains = {
+  1: 'mainnet',
+  10: 'optimism-mainnet',
+  5: 'goerli',
+  420: 'optimism-goerli',
+} as const
+
+if (!glob.sync('node_modules/*').length) {
+  throw new Error(
+    'No node_modules found. Please run `pnpm install` before running this script'
+  )
+}
+
+const nodeModules = path.join(__dirname, '..', 'node_modules')
+
+const bedrockDeployments = {
+  [1]: glob.sync(
+    path.join(nodeModules, '@eth-optimism', 'contracts-bedrock', 'deployments', chains[1], '*.json')
```

Comment:
Why is this going thru node modules instead of the local package?

Reply:
Touching code outside of your own package in an unexplicit way in a package based monorepo is a bad practice. When you go through node modules you are guaranteeing you correctly declared your dependency instead of it being implicit